### ISSUE TITLE
 INSTALL FAILED with PhaseScriptExecution

### ISSUE THREAD

**Issue by NilStack** [realm/SwiftLint#158]
** INSTALL FAILED **

While making install, i got INSTALL FAILED message:

The following build commands failed:
	PhaseScriptExecution Run\ SwiftLint /Users/Peng/Library/Developer/Xcode/DerivedData/SwiftLint-hhenhvsyjwggxiebhwkoovnebxbj/Build/Intermediates/ArchiveIntermediates/swiftlint/IntermediateBuildFilesPath/SwiftLint.build/Release/swiftlint.build/Script-E819854B1B09A3CB00CEB0D9.sh
(1 failure)
make: *** [installables] Error 65

any idea? Thanks

**Comment by keith** [comment]
Are you running Xcode 7.x?

**Comment by NilStack** [comment]
Yes, release version : Version 7.0 (7A218).

**Comment by d-ronnqvist** [comment]
Same problem on Version 7.1.1 (7B1005) from the MAS

**Comment by jpsim** [comment]
@NilStack @d-ronnqvist sorry you're hitting this, I'm not sure what's happening. It would help if I knew more about how you triggered this error. Are you building SwiftLint from source by running `make install`? If not, how did you trigger this error?

What do the following commands print for you?

```shell
$ ls /Applications | grep Xcode
$ xcode-select -p
$ xcodebuild -version
```

Any other steps you could share that would help me reproduce this issue would be appreciated.

Thanks!

**Comment by NilStack** [comment]
Hi, @jpsim Thanks for the response. 
I tried the latest codes and got the same results.
The steps to trigger this error are
1. git clone
2. git submodule update --init --recursive
3. make install

The results for the commands in your list are:

    $ ls /Applications | grep Xcode
          Xcode.app
    $ xcode-select -p
         /Applications/Xcode.app/Contents/Developer
    $ xcodebuild -version
        Xcode 7.1.1
        Build version 7B1005

Thanks

**Comment by jpsim** [comment]
The reason `make install` is failing is that the `Run SwiftLint` run script build phase is failing. This shouldn't be happening because the script *should* handle `swiftlint` not being installed:

```shell
if which swiftlint >/dev/null; then
  swiftlint
else
  echo "SwiftLint does not exist, download from https://github.com/realm/SwiftLint"
fi
```

**Comment by d-ronnqvist** [comment]
My steps are the same (git clone, git submodule update --init --recursive, make install)

```
$ ls /Applications | grep Xcode
    Xcode.app
$ xcode-select -p
    /Applications/Xcode.app/Contents/Developer
$ xcodebuild -version
    Xcode 7.1.1
    Build version 7B1005
```

If it's relevant: I have installed it from source before, and am now trying to update by installing a newer version.

**Comment by d-ronnqvist** [comment]
My problem was resolved by going in and manually removing `/usr/local/bin/swiftlint` and then running `make install` again (at which point it installed)

**Comment by jpsim** [comment]
@d-ronnqvist did SwiftLint work before you ran `make install`? I suspect that this error would occur if SwiftLint had linker errors (causing `which swiftlint` to succeed, but `swiftlint` to fail).

**Comment by d-ronnqvist** [comment]
It used to run, but was an older version. We upgraded for the configurability support and some of us (those not using brew) encountered this issue.

**Comment by jpsim** [comment]
Ah, this would also happen if SwiftLint (or SourceKit) crashes, which is likely when using it on Swift files using a later version of the language.

I think @neonichu has a way to detect the Swift version used in a file, we could use that to fail more elegantly if it's a later version than SwiftLint supports.

**Comment by jpsim** [comment]
This should be fixed as of #215.
